src/masternodeman.cpp

```diff
@@ -855,7 +860,15 @@ int CMasternodeMan::ProcessMessageInner(CNode* pfrom, std::string& strCommand, C
 
     if (strCommand == NetMsgType::MNBROADCAST) {
         CMasternodeBroadcast mnb;
-        vRecv >> mnb;
+        // Use addr v2 format after v5.3 upgrade.
+        int isActiveV5_3 = Params().GetConsensus().NetworkUpgradeActive(GetBestHeight(), Consensus::UPGRADE_V5_3);
+        if (isActiveV5_3) {
+            OverrideStream<CDataStream> s(&vRecv, vRecv.GetType(), vRecv.GetVersion() | ADDRV2_FORMAT);
+            s >> mnb;
+        } else {
+            // Pre-v5.3 serialization format
+            vRecv >> mnb;
+        }
         return ProcessMNBroadcast(pfrom, mnb);
```

Comment:
Mhm... this could be problematic.
If nodes start processing `MNBROADCAST` messages with new serialization after v5.3 upgrade, this means that old (pre-v5.3) mnb messages might not be properly parsed at that point.

So, with this, ALL masternodes would have to send *another* start message after enforcement (in a protocol upgrade, the new mnb is sent with the new client *at any time before* the enforcement block).